### ISSUE TITLE
Email transport needs TLS trust settings

### ISSUE THREAD

**Issue by ltning** [Graylog2/graylog2-server#7711]
Each in- or outbound connection that supports TLS needs a full complement of setttings to be specified, particularly the certificates that are to be trusted. The CA I trust for our elasticsearch cluser is not necessarily the same as for e.g. SMTP servers.

## Expected Behavior
For each kind of system-level in- or outbound connection (https, elasticsearch back-end, email, etc.), it must be possible to specify different truststores or trusted certificate files.

## Current Behavior
The system-wide truststore is used for all connections, which is not desirable and in fact significantly weakens protection, particularly on the elasticsearch and mongodb back-end connections.

## Context
There are serious issues with the way certificate trust is implemented. As it seems, the only way to configure trusted certificates is to use a system-wide truststore, defined at startup. This is a major concern since the certificates presented by our elasticsearch cluster (for example) are internal only, and we specifically do not want to trust anything else for those connections. The problem rears its head when trying to configure SSL or TLS for email; the same truststore is being used but the email server is presenting a commercially-issued certificate, which is not in the truststore. In addition, creating the dependency of having to modify the truststore whenever the email server changes certificate authority _will_ cause outages, and one is not necessarily notified when this happens.

In other words: For each purpose, a separate truststore setting is needed - much like each input can have its own TLS certificate and trusted certificates specified.

## Your Environment
* Graylog Version: 3.1.3
* Operating System: FreeBSD 12.1
